strings/validator/validator.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package validator defines various validator functions. The functions
+// are mainly intended to be used in strings/slices functions.
+package validator
+
+import (
+	"net"
+)
+
+// IP returns a validator for net.IP. IPVersion=0 matches both families
+func IP(IPVersion uint) func(string) bool {
```

Comment:
we have k/utils/net `IsIPv6String()` and `IsIPv4String` (ditto CIDRString).  We could add `IsIPString` (which is just `net.ParseIP() != nil`) and then we don't need this validator package at all.

Reply:
Agree. I re-wrote the Example with `IsIPv6String()` and it looks much better.